File: server/java/EnvLoader.java
import java.io.File;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

/**
 * Utility class for loading environment variables from .env file.
 * Uses manual parsing to avoid dependency on dotenv-java at runtime.
 */
public class EnvLoader {
    private static Map<String, String> cachedEnv = null;
    
    /**
     * Loads environment variables from .env file in server directory.
     */
    public static Map<String, String> loadEnv() {
        if (cachedEnv != null) {
            return cachedEnv;
        }
        
        Map<String, String> env = new HashMap<>();
        
        // Get the parent directory (server/) where .env file is located
        Path currentPath = Paths.get(System.getProperty("user.dir"));
        Path envPath = currentPath.getParent().resolve(".env");
        
        File envFile = envPath.toFile();
        
        if (envFile.exists()) {
            try (Scanner scanner = new Scanner(envFile)) {
                while (scanner.hasNextLine()) {
                    String line = scanner.nextLine().trim();
                    
                    // Skip empty lines and comments
                    if (line.isEmpty() || line.startsWith("#")) {
                        continue;
                    }
                    
                    // Parse key=value pairs
                    int equalsIndex = line.indexOf('=');
                    if (equalsIndex > 0) {
                        String key = line.substring(0, equalsIndex).trim();
                        String value = line.substring(equalsIndex + 1).trim();
                        
                        // Remove quotes if present
                        if (value.startsWith("\"") && value.endsWith("\"") && value.length() > 1) {
                            value = value.substring(1, value.length() - 1);
                        } else if (value.startsWith("'") && value.endsWith("'") && value.length() > 1) {
                            value = value.substring(1, value.length() - 1);
                        }
                        
                        env.put(key, value);
                    }
                }
            } catch (Exception e) {
                System.err.println("Warning: Could not read .env file: " + e.getMessage());
            }
        } else {
            System.err.println("Warning: .env file not found at: " + envPath);
        }
        
        cachedEnv = env;
        return env;
    }
    
    /**
     * Gets an environment variable, with fallback to system environment.
     */
    public static String getEnv(String key, String defaultValue) {
        Map<String, String> env = loadEnv();
        String value = env.get(key);
        if (value == null || value.isEmpty()) {
            value = System.getenv(key);
        }
        return (value != null && !value.isEmpty()) ? value : defaultValue;
    }
    
    /**
     * Gets an environment variable.
     */
    public static String getEnv(String key) {
        return getEnv(key, null);
    }
}

File: server/kotlin/EnvLoader.kt
import java.io.File
import java.nio.file.Paths

/**
 * Utility object for loading environment variables from .env file.
 */
object EnvLoader {
    private var cachedEnv: Map<String, String>? = null
    
    /**
     * Loads environment variables from .env file in server directory.
     */
    fun loadEnv(): Map<String, String> {
        if (cachedEnv != null) {
            return cachedEnv!!
        }
        
        val env = mutableMapOf<String, String>()
        
        // Get the parent directory (server/) where .env file is located
        val currentPath = Paths.get(System.getProperty("user.dir"))
        val envPath = currentPath.parent.resolve(".env")
        val envFile = envPath.toFile()
        
        if (envFile.exists()) {
            try {
                envFile.readLines().forEach { line ->
                    val trimmed = line.trim()
                    
                    // Skip empty lines and comments
                    if (trimmed.isEmpty() || trimmed.startsWith("#")) {
                        return@forEach
                    }
                    
                    // Parse key=value pairs
                    val equalsIndex = trimmed.indexOf('=')
                    if (equalsIndex > 0) {
                        val key = trimmed.substring(0, equalsIndex).trim()
                        var value = trimmed.substring(equalsIndex + 1).trim()
                        
                        // Remove quotes if present
                        if (value.startsWith("\"") && value.endsWith("\"") && value.length > 1) {
                            value = value.substring(1, value.length - 1)
                        } else if (value.startsWith("'") && value.endsWith("'") && value.length > 1) {
                            value = value.substring(1, value.length - 1)
                        }
                        
                        env[key] = value
                    }
                }
            } catch (e: Exception) {
                System.err.println("Warning: Could not read .env file: ${e.message}")
            }
        } else {
            System.err.println("Warning: .env file not found at: $envPath")
        }
        
        cachedEnv = env
        return env
    }
    
    /**
     * Gets an environment variable, with fallback to system environment.
     */
    fun getEnv(key: String, defaultValue: String = ""): String {
        val env = loadEnv()
        val value = env[key]
        if (value == null || value.isEmpty()) {
            val systemValue = System.getenv(key)
            return systemValue ?: defaultValue
        }
        return value
    }
}

